arangod/RocksDBEngine/RocksDBCollection.cpp

```diff
@@ -1088,6 +1106,7 @@ Result RocksDBCollection::performUpdateOrReplace(transaction::Methods* trx,
   res = lookupDocumentVPack(trx, oldDocumentId, previousPS,
     /*readCache*/ true, /*fillCache*/ false, ReadOwnWrites::yes);
   if (res.fail()) {
+    ::reportPrimaryIndexInconsistency(res, keyStr, oldDocumentId);
```

Comment:
Just mentioning that reporting an error here could be a false positive in case we don't have a snapshot.
But I think all write operations will acquire a snapshot when the operation/transaction starts, so we should be fine.